src/decomposition.jl

```diff
@@ -0,0 +1,167 @@
+"""
+    ear_decomposition(g, s)
+
+Return an Ear decomposition of the graph by using s as source for DFS Tree.
+
+An ear of an undirected unweighted connected graph `G` is a path `P` where
+the two endpoints of the path may coincide (i.e., form a cycle), but where
+otherwise no repetition of edges or vertices is allowed, so every internal
+vertex of `P` has degree two in `P`.
+
+An ear decomposition of an undirected graph `G` is a partition of its
+set of edges into a sequence of ears, such that the one or two endpoints
+of each ear belong to earlier ears in the sequence and such that the
+internal vertices of each ear do not belong to any earlier ear.
```

Comment:
Isn't this sentence nearly exactly the one on Wikipedia? We might get into trouble if we copy such things, so you might have to write that in your own words.

Reply:
Yes, it's from Wikipedia, I thought writing own will not explain clearly as compared to Wikipedia.

Ear Decomposition: Partition the edge set of undirected graph `G` into a
sequence of paths(ears), such that only the end vertices of each path are
present in earlier paths. 

Does this make sense?

Reply:
Shouldn't that then be "sequence of paths or cycles (ears)"?